### PR TITLE
Bugfix: handle errors where the `details` field is returned as an object instead of an array of objects

### PR DESCRIPTION
**Pull request by manicminer**

### CODE REVIEW

**Review by mbfrahry — APPROVED** [review]
LGTM!

**Review by tombuildsstuff — CHANGES_REQUESTED** [review]

File: sdk/odata/errors.go [hunk 1]
```diff
@@ -77,6 +80,20 @@ func (e *Error) UnmarshalJSON(data []byte) error {
...
+		if string((*raw)[0]) != "[" {
+			// When `details` is not an array, surround with square brackets to make it so
+			*raw = append([]byte{0x5b}, append(*raw, 0x5d)...)
+		}
+		var details ErrorDetails
+		if err := json.Unmarshal(*raw, &details); err != nil {
+			return err
+		}
+		e.Details = &details
+	}
```
[the remark this answers is not in the document]
Comment (manicminer) [reply sdk/odata/errors.go:95]: Sure thing 👍

[the remark this answers is not in the document]
Comment (tombuildsstuff) [reply sdk/odata/errors.go]: Rebased this / pushed a commit to conditionally try deserialising this into both types without parsing the response fwiw

**Review by tombuildsstuff — APPROVED** [review]

File: sdk/odata/errors.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (manicminer) [reply sdk/odata/errors.go]: Looks good, thanks for that 👍

**Pull request merged**